src/settings.cpp

```diff
@@ -316,20 +320,45 @@ CriticalSection g_csConfigValues;
 } // anonymous namespace
 
 
-void Settings::DoWriteConfigValue(const char *name, const wchar_t *value)
+int __cdecl Settings::DefaultConfigRead(void *, const char *name, wchar_t* buf, size_t len)
 {
-    CriticalSectionLocker lock(g_csConfigValues);
+    //RegistryRead requires array size in bytes
+    return RegistryRead(name, buf, len * sizeof(wchar_t));
+}
 
+int __cdecl Settings::DefaultConfigWrite(void *, const char *name, const wchar_t* value)
```

Comment:
Actually the more important comment here is that these very thin wrappers don't serve a useful role — they just trivially call another *private* function, so let's just refactor the private functions to be directly usable.